### PR TITLE
Cors/improvement

### PR DESCRIPTION
**Pull request by axblueblader**
- Refactor to apply defaults for missing fields
- Attempt to improve performance by using more constants and less Streams

### CODE REVIEW

File: middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java [hunk 1]
```diff
@@ -112,7 +141,7 @@ private static void handlePreflight(Context ctx, @NonNull Config config) {
         }
     }
 
-    private static void handleActual(Context ctx, Config config) {
+    private static void handleActual(Context ctx,@NonNull Config config) {
```
Comment (huynhminhtufu) [inline middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java:144]: Missed a space here, should be `Context ctx, @NonNull Config`

**Comment by huynhminhtufu** [comment]
@axblueblader Please consider label it as a Draft pull request when it's WIP, otherwise, the PRs can be reviewed and merged.
image

**Comment by axblueblader** [comment]
ok, I'll remember to do it next time. This one is available to be reviewed and merged

File: middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java [hunk 2]
```diff
@@ -136,20 +165,41 @@ private static void handleActual(Context ctx, Config config) {
...
+    private static boolean isAllowedHeaders(String[] reqHeaders, List<String> allowHeaders) {
+        if (reqHeaders[0].isEmpty()) {
             return true;
         }
-        return reqHeaders.stream()
-                .allMatch(header -> config.allowHeaders
-                        .stream()
-                        .anyMatch(header::equalsIgnoreCase));
+        for (String header : reqHeaders) {
+            boolean allow = false;
```
Comment (huynhminhtufu) [inline middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java:191]: Please use `isSomething` for `boolean` variable, it's more declarative
Expected: `allow` -> `isAllowed`

File: middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java [hunk 3]
```diff
@@ -136,20 +165,41 @@ private static void handleActual(Context ctx, Config config) {
         }
     }
 
-    private static boolean isAllowedOrigin(String origin, Config config) {
-        return config.allowOrigins.stream().anyMatch(org -> org.equalsIgnoreCase(origin));
+    private static boolean isAllowedMethod(String method, List<String> allowMethods) {
+        for (String am : allowMethods) {
```
Comment (huynhminhtufu) [inline middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java:169]: Can it be rewritten to `val am : allowMethods`

Comment (huynhminhtufu) [reply middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java:169]: Btw, IMHO we should use `lombok.var` and `lombok.val` where possible, which will make our code more consistency, I'll add this to the Developer Guide

File: middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java (hunk 2 above)
Comment (huynhminhtufu) [reply middlewares/src/main/java/com/jinyframework/middlewares/cors/Cors.java:191]: Please consider using` lombol.var`

**Review by huynhminhtufu — APPROVED** [review]
LGTM

**Pull request merged**